project/conventions.md

```diff
@@ -0,0 +1,108 @@
+the strict definitions user RFC 2119, but SHOULD be interpreted in
+context with a meaning aligned to RFC 2119.
+
+## Helm Chart conventions
+
+Helm values are organized in a tree structure, with each entry having
+a key, the path within the structure, and a value, the contents held
+at that path.  Some of the contents include objects and arrays, which
+are structures that permit multiple values under a particular path.
+
+This design choice mimics the representation of a file system, and we
+may use file system analogies to describe the conventions and choices
+in the project.
+
+### Preference for absolute paths in templates
+
+Outside of looping constructs that require relative paths, templates
+should use absolute paths (paths starting with the `.` character) for
+their value references.  
+
+### Helm Key path elements
+
+Elements refer to items in the configuration.  Some of those items are
+external to the project, while others are components the project choose
+to implement the Helm Chart packaging solution.
+
+### Preferred characters for helm path elements
+
+Elements within a helm chart path SHOULD limit the use of special
+characters, even when such characters are supported by Helm.  ASCII
+letters, upper and lower case, ASCII numerals, and the ASCII underscore
+character are always permitted.  
+
+Other characters, such as `@`, `~`, `-`, `&`, `+` are often used as
+shorthand to represent, respectively `at`, `similar / not`, 
+`minus / without`, `and / reference`, `plus / with`.  Helm key elements
+MUST NOT contain such symbols, as it burdens the reader to deduce the
+intent of the writer.
+
+### Camel Case for created elements.
+Elements that refer to components that the project has created should
+exclusively use camel case, with a the first word being lower case.  To
+illustrate, a "card sort order" path element would read "cardsharper".
+
+### Acronyms and Minimalists
+
+Occasionally the use of an Acronym or Minimalist is used as part of an
+element name.  An acronym is the use of the first letter of a phrase
+or multi-word proper name that is pronounceable, while an initialisim is
+the use of the first letter of a phrase or multi-word proper name 
+pronounced by naming the letters.  `NASA` is an example of an acronym,
+while `NFL` is an example of an Initialisim.
+
+To keep the recognition of the Acronym or Initialisim, an all capital
+representation of the Acronym or Initialisim MUST be maintained.  To
+reduce the confusion this can create in camel case elements, we recommend
+that such Acronyms and Initialisim be placed at the end of the element
+or the element is restructured to avoid confusion.  For example, 
+`mailingListNASA` or `mailingList.NASA` is preferred over
+`NASAMailingList`.
+
+Acronyms and Initialisim occasionally use periods in their presentation.
+When they do, the periods MUST be removed without replacement by
+another character.  To illustrate `U.S.S.R.` should be written as
+`USSR`.
```

Comment:
Suggest not to use USSR as it no longer exists.

Reply:
Sure, give me another example of an Acronym that is typically spelled with periods.

And I'm comfortable referring to stuff that existed in the past, as long as it's not in a context that implies it exists today (when it doesn't); but, I really would like a better example too.  I just can't think of a common one to illustrate the point.

Reply:
U.S., U.N. as United Nations